crates/viewer/re_space_view_spatial/src/max_image_dimension_subscriber.rs

```diff
@@ -89,62 +91,75 @@ impl ChunkStoreSubscriber for MaxImageDimensionSubscriber {
-                                reader.set_format(format);
-                            }
-                        }
-
-                        if let Ok((width, height)) = reader.into_dimensions() {
-                            let max_dim = self
-                                .max_dimensions
-                                .entry(event.store_id.clone())
-                                .or_default()
-                                .0
-                                .entry(event.diff.chunk.entity_path().clone())
-                                .or_default();
-
-                            max_dim.height = max_dim.height.max(height);
-                            max_dim.width = max_dim.width.max(width);
-                        }
-                    }
+                if let Some([width, height]) = size_from_blob(blob.as_ref(), media_type.as_deref())
+                {
+                    let max_dim = self
+                        .max_dimensions
+                        .entry(event.store_id.clone())
+                        .or_default()
+                        .0
+                        .entry(event.diff.chunk.entity_path().clone())
+                        .or_default();
+                    max_dim.width = max_dim.width.max(width);
+                    max_dim.height = max_dim.height.max(height);
                 }
             }
         }
     }
 }
+
+fn size_from_blob(blob: &dyn Array, media_type: Option<&dyn Array>) -> Option<[u32; 2]> {
+    re_tracing::profile_function!();
+
+    let blob = Blob::from_arrow_opt(blob).ok()?.first()?.clone()?;
+
+    let media_type: Option<MediaType> = media_type
+        .and_then(|media_type| MediaType::from_arrow_opt(media_type).ok())
+        .and_then(|list| list.first().cloned())
+        .flatten();
+
+    let media_type = MediaType::or_guess_from_data(media_type, &blob)?;
+
+    if media_type.is_image() {
+        re_tracing::profile_scope!("image");
+
+        let image_bytes = blob.0.as_slice();
+
+        let mut reader = image::io::Reader::new(std::io::Cursor::new(image_bytes));
+
+        if let Some(format) = image::ImageFormat::from_mime_type(&media_type.0) {
+            reader.set_format(format);
+        }
+
+        if reader.format().is_none() {
+            if let Ok(format) = image::guess_format(image_bytes) {
+                // Weirdly enough, `reader.decode` doesn't do this for us.
+                reader.set_format(format);
+            }
+        }
+
+        reader.into_dimensions().ok().map(|size| size.into())
+    } else if media_type.is_video() {
+        re_tracing::profile_scope!("video");
+        if true {
+            None // TODO(#7821): Use the VideoCache here so we make sure we only load each video ONCE
+        } else {
+            re_video::VideoData::load_from_bytes(&blob, &media_type)
+                .ok()
+                .map(|video| video.dimensions())
+        }
```

Comment:
This is the core of it. I'll work on having access to `VideoCache` here, and then we can resolve this in a follow-up PR.

…or we add functionality to `re_mp4` to very quickly parse an .mp4 to only look for its size info 🤔